src/S3.ts

```diff
@@ -16,43 +16,45 @@ class S3 {
   }
 
   public async initialise() {
-    const s3UrlBase: any = new S3File.Endpoint(this.url)
-    this.s3Handler = new S3File.S3({
-      endpoint: s3UrlBase,
-      accessKeyId: this.params.keyId,
-      secretAccessKey: this.params.secretAccessKey,
-      s3ForcePathStyle: s3UrlBase.protocol === 'http:',
+    const s3UrlBase: any = new URL(this.url)
+    this.s3Handler = new S3Client({
+      credentials: {
+        accessKeyId: this.params.keyId,
+        secretAccessKey: this.params.secretAccessKey,
+      },
+      endpoint: s3UrlBase.href,
+      forcePathStyle: s3UrlBase.protocol === 'http:',
     })
-    try {
-      if ((await this.bucketExists(this.bucketName)) === true) {
-        return true
-      }
-    } catch (err) {
-      throw new Error(`Unable to connect to S3, either S3 login credentials are wrong or bucket cannot be found
+
+    return this.bucketExists(this.bucketName)
+      .then(() => true)
+      .catch(() => {
+        throw new Error(`Unable to connect to S3, either S3 login credentials are wrong or bucket cannot be found
                             Bucket used: ${this.bucketName}
                             End point used: ${s3UrlBase.href}
-                            Public IP used: ${await publicIpv4()}`)
-    }
+                            Public IP used: ${publicIpv4()}`)
+      })
   }
 
-  public async bucketExists(bucket: string): Promise<any> {
+  public bucketExists(bucket: string): Promise<any> {
```

Comment:
You have removed the `async` here, what is the rational behind?

Reply:
no particular reason seemed a little redundant here.

Reply:
Not sure myself, but this code has to apply asynchrnously, so redundant with which part of the code exactl?